src/Rector/Rector/Namespace_/AppUsesStaticCallToUseStatementRector.php

```diff
@@ -67,7 +67,9 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        $this->nodeRemover->removeNodes($appUsesStaticCalls);
+        foreach($appUsesStaticCalls as $toBeRemovedNode) {
+            $this->nodeRemover->removeNode($toBeRemovedNode);
+        }
```

Comment:
`NodeRemover` will no longer exists in next Rector release, better to change with traverser instead:

```php
$this->traverseNodesWithCallable($node->stmts, function (Node $subNode) use ($appUsesStaticCalls) {
       foreach ($appUsesStaticCalls as $toBeRemovedNode) {
               if ($toBeRemovedNode === $subNode) {
                    return \PhpParser\NodeTraverser::REMOVE_NODE;
               }
       }
});
```

Reply:
Thank you for that notice and the fix!
I hope rector will have some sort of upgrade guide / doc where plugin devs get information on which old behavior can be replaced with which new functionality like you did here 😅

Reply:
/cc @tomasVotruba

Reply:
@samsonasik 
Well in my case i still have some problems here.
Just replacing the code leads to the following error:
```
LogicException: enterNode() returned invalid value of type integer
LogicException: enterNode() returned invalid value of type integer
/mypath/vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:128
/mypath/vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:196
/mypath/vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:85
/mypath/vendor/rector/rector/packages/PhpDocParser/NodeTraverser/SimpleCallableNodeTraverser.php:27
/mypath/vendor/rector/rector/src/Rector/AbstractRector.php:294
/mypath/src/Rector/Rector/Namespace_/AppUsesStaticCallToUseStatementRector.php:78
```

Reply:
I am from mobile so probably missing something. You may look current stmt in case of it under stmt that can't directly be removed, eg: inside If_->cond. 

Could you give me write access to this branch for a while? 

Otherwise, I probably need to create a PR to your PR

Reply:
Ok, I got it, I created PR to your PR branch:

-  https://github.com/cakephp/upgrade/pull/236

Reply:
Thank you. I leave this up to our lead dev since I am pretty new to the whole AST and PHP Parser system.